### PR TITLE
[FE] [MODIFY] Eip 230 resto app profile

### PR DESCRIPTION
**Pull request by wernerramon**
### 🛠 Changes being made

Updated Profile for the restaurant app. Please test together with PR on Guardos (EIP-230) (https://github.com/GuardosEpitech/Guardos/pull/55) to test all functions

### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: restoMobile/src/pages/ProfileScreen/ChangePassword/ChangePasswordScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+({navigation}) => {
+  const [oldPassword, setOldPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+
+  const handleSave = () => {
+    // Handle save logic here
+    console.log('Old Password:', oldPassword);
+    console.log('New Password:', newPassword);
+    console.log('Confirm Password:', confirmPassword);
```
Comment (JosefineM2000) [inline restoMobile/src/pages/ProfileScreen/ChangePassword/ChangePasswordScreen.tsx:28]: please remove those logs

File: restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  ];
+
+  useEffect(() => {
+    fetchUserData();
+  }, []);
+
+  const fetchUserData = async () => {
+    try {
+      const token = await AsyncStorage.getItem('userToken')
+      const response = await fetch(baseUrl + 'data', {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx:49]: BE calls belong in service files, not in the component directly

File: restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+      }
+      const userData = await response.json();
+      setUsername(userData.username);
+      setEmail(userData.email);
+      setLocation(userData.location);
+    } catch (error) {
+      console.error('Error fetching user data:', error);
+    }
+  };
+    const selectImage = async () => {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx:67]: please fix indentation and 1 free line between all methods

File: restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+            },
+          },
+        ],
+        { cancelable: false }
+      );
+  };
+
+  const handleApplyChanges = async () => {
+    try {
+      const response = await fetch(baseUrl + 'update', {
```
Comment (JosefineM2000) [inline restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx:120]: again, BE calls should be in file under services folder

File: restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+      console.error('Error updating user data:', error);
+    }
+  };
+
+  const handleNavigateToChangePassword = () => {
+    navigation.navigate('Change Password');
+  };
+
+  return (
+    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
```
Comment (JosefineM2000) [inline restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx:148]: screen should be scrollable in case content exceeds 100% screen height.

File: restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+            setValue={setLanguage}
+        />
+       <View style={styles.buttonContainer}>
+          <Button 
+            title="Apply Changes" 
+            onPress={handleApplyChanges} color="green" 
+          />
+        </View>
+        <View style={styles.logoutButtonContainer}>
+          <Button title="Logout" onPress={handleLogout} color="red" />
```
Comment (JosefineM2000) [inline restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx:202]: please use our Guardos red instead and place the logout button below all content instead of having an absolute position

File: restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+        />
+        <View style={styles.changePasswordButton}>
+            <Button 
+              title="Change Password" 
+              onPress={handleNavigateToChangePassword} 
+            />
+        </View>
+        <TextInput
+          style={styles.input}
+          placeholder="Location"
```
Comment (JosefineM2000) [inline restoMobile/src/pages/ProfileScreen/Profile/NewProfile.tsx:184]: for resto, we have menu design (dropdownpicker like language) instead of location

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]
